### ISSUE TITLE
connecting signal in the UI will result in a code connect signal function

### ISSUE THREAD

**Issue by ywmaa** [godotengine/godot-proposals#5959]
### Describe the project you are working on

any godot project

### Describe the problem or limitation you are having in your project

I like to have clean code + obvious code, so we have two ways to connect signals in godot, use the UI signals tab, or connect using code, I chose the second option because if I read the code or someone else, he/I will immediatly understand what is the connected function do, and it is connected to which node.

 now if someone does a project in godot and shares it online it can have both the ways used, or using only UI connector, which atleast for me makes the logic of the code hard to understand, unless I look a lot between the script editor and the signals tab + scene tree.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

we shouldn't remove the signals tab, it is very useful, but instead we can make when we connect it to a script, it generates a node.connect("signal name",signal_function), inside the  _ready() or _init() the best one for this kind of setup.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

N/A

### If this enhancement will not be used often, can it be worked around with a few lines of script?

of course it can be worked around by just declaring signals only in code.

### Is there a reason why this should be core and not an add-on in the asset library?

I think this is an editor thing, and I am not sure if we can make a plugin for this, plus the idea is all about sharing projects, so whatever the way you go inside the editor the result will be the same. so having this as an add-on will not make it a standard.

**Comment by ywmaa** [comment]
also sorry about the title I couldn't make it an easy to understand title

**Comment by Calinou** [comment]
You may not have any scripts at all in the current scene, so this approach wouldn't be viable in every case.

**Comment by ywmaa** [comment]
> You may not have any scripts at all in the current scene, so this approach wouldn't be viable in every case.

create a script for it ?, plus I am not sure when we will connect a signal without having a script or going to add a script later ?

**Comment by KoBeWi** [comment]
I have many scenes that use a generic script and some of the scenes have signal connections that should not be in the script (because the same script can be used in scenes that don't need these connections).

**Comment by YuriSizov** [comment]
I agree with the principle of this proposal, though. I don't like to have signal connections in scene files, that breaks logic between the code that you can read, and the scene files that you can't read (but can inspect in other ways, getting distracted from reading the code). It's nice to have zero-code approach available, but when it gets mixed with code-first approach, you get a mess. So I think it would be nice to have this in some way. Especially if we end up implementing a signal connect annotation.

Can be a project-wide setting, which would switch between using scenes and using code, and if you opt to use code and don't have a script it would err. For shared scripts that don't need to share signals I don't have a suggestion, but I find that whole situation to be quite esoteric.
